WordPress/Sniffs/VIP/RestrictedVariablesSniff.php

```diff
@@ -65,4 +88,35 @@ public function getGroups() {
 		);
 	}
 
+	/**
+	 * This sniff is active, but it's deprecated, handle the deprecation notices
+	 *
+	 * @since 1.0.0 Added to allow for throwing the deprecation notices.
+	 *
+	 * @param int $stackPtr The position of the current token in the stack.
+	 *
+	 * @return void|int
+	 */
+	public function process_token( $stackPtr ) {
+		if ( false === $this->thrown['DeprecatedSniff'] ) {
+			$this->thrown['DeprecatedSniff'] = $this->phpcsFile->addWarning(
+				'The "WordPress.VIP.RestrictedVariablesSniff" sniff has been deprecated. Please update your custom ruleset.',
+				0,
+				'DeprecatedSniff'
+			);
+		}
+
+		if ( ! empty( $this->exclude )
```

Comment:
A unit test should be added to test that the deprecated property warning is thrown correctly.